### PR TITLE
[android] Tidy up annotations API

### PR DESCRIPTION
**Pull request by ljbade**
Make unimplemented properties package private.
Make annotations object properties immutable.
Make InfoWindow and Circle classes package private.
Mininal version of #2546 suitable for v2.1.0 release.

### CODE REVIEW

File: android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Annotation.java [hunk 1]
```diff
@@ -12,20 +12,23 @@
...
 
+    /**
+     * Do not use this method. Used internally by the SDK.
+     */
     public long getId() {
         return id;
     }
 
-    public boolean isVisible() {
+    boolean isVisible() {
```
Comment (bleege) [inline android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Annotation.java]: Why remove `public`?

File: android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Annotation.java [hunk 2]
```diff
@@ -34,19 +37,25 @@ public void remove() {
         mapView.removeAnnotation(this);
     }
 
-    public void setAlpha(float alpha) {
+    void setAlpha(float alpha) {
```
Comment (bleege) [inline android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Annotation.java]: Why remove `public`?

File: android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Annotation.java [hunk 3]
```diff
@@ -34,19 +37,25 @@ public void remove() {
...
 
+    /**
+     * Do not use this method. Used internally by the SDK.
+     */
     public void setMapView(MapView mapView) {
         this.mapView = mapView;
     }
 
-    public void setVisible(boolean visible) {
+    void setVisible(boolean visible) {
```
Comment (bleege) [inline android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Annotation.java]: Why remove `public`?

**Comment by ljbade** [comment]
@bleege Core GL does not currently implement the `visible` or `isVisible` properties.
`setAlpha` on an existing `Polyline`/`Polygon` object will not update the marker already on the map and will confuse developers. It is only needed for the builder pattern in `PolylineOptions`/`PolygonOptions`.

Also I just remembered that `Marker` does not support `alpha` at all in Core GL. Will remove from `MarkerOptions`.

**Comment by ljbade** [comment]
Also made methods `private` instead of package private where I could.

**Pull request merged**